src/cloudfront.rs

```diff
@@ -1,45 +1,50 @@
-use anyhow::Context;
-use aws_sigv4::http_request::{self, SignableRequest, SigningSettings};
-use aws_sigv4::SigningParams;
-use reqwest::blocking::Client;
+use aws_credential_types::Credentials;
+use aws_sdk_cloudfront::config::Region;
+use aws_sdk_cloudfront::error::SdkError;
+use aws_sdk_cloudfront::types::{InvalidationBatch, Paths};
+use aws_sdk_cloudfront::{Client, Config};
 use retry::delay::{jitter, Exponential};
 use retry::OperationResult;
-use secrecy::{ExposeSecret, SecretString};
-use std::time::{Duration, SystemTime};
+use std::panic::AssertUnwindSafe;
+use std::time::Duration;
+use tokio::runtime::Runtime;
 
-#[derive(Clone)]
 pub struct CloudFront {
+    client: AssertUnwindSafe<Client>,
     distribution_id: String,
-    access_key: String,
-    secret_key: SecretString,
 }
 
 impl CloudFront {
     pub fn from_environment() -> Option<Self> {
         let distribution_id = dotenvy::var("CLOUDFRONT_DISTRIBUTION").ok()?;
         let access_key = dotenvy::var("AWS_ACCESS_KEY").expect("missing AWS_ACCESS_KEY");
-        let secret_key = dotenvy::var("AWS_SECRET_KEY")
-            .expect("missing AWS_SECRET_KEY")
-            .into();
+        let secret_key = dotenvy::var("AWS_SECRET_KEY").expect("missing AWS_SECRET_KEY");
+
+        let credentials = Credentials::from_keys(access_key, secret_key, None);
+
+        let config = Config::builder()
+            .region(Region::new("us-east-1"))
+            .credentials_provider(credentials)
+            .build();
+
+        let client = AssertUnwindSafe(Client::from_conf(config));
 
         Some(Self {
+            client,
             distribution_id,
-            access_key,
-            secret_key,
         })
     }
 
     /// Invalidate a file on CloudFront
     ///
     /// `path` is the path to the file to invalidate, such as `config.json`, or `re/ge/regex`
-    #[instrument(skip(self, client))]
-    pub fn invalidate(&self, client: &Client, path: &str) -> anyhow::Result<()> {
-        let path = path.trim_start_matches('/');
-        let url = format!(
-            "https://cloudfront.amazonaws.com/2020-05-31/distribution/{}/invalidation",
-            self.distribution_id
-        );
-        trace!(?url);
```

Comment:
Seems some trace logging has been removed. Is it intentional?

Reply:
with the SDK we don't build the URL ourselves anymore so we can't trace log it

Reply:
Ah I see, thanks for clarifying!